lib/user-interface/react-app/src/app.tsx

```diff
@@ -18,9 +18,10 @@ import MultiChatPlayground from "./pages/chatbot/playground/multi-chat-playgroun
 import RssFeed from "./pages/rag/workspace/rss-feed";
 
 function App() {
+
   return (
     <div style={{ height: "100%" }}>
-      <BrowserRouter>
+      <HashRouter>
```

Comment:
What's the reasoning behind switching to `HashRouter` ?

Reply:
BrowserRouter isn't very [compatible with direct S3 website hosting.](https://stackoverflow.com/questions/66326106/react-router-locations-on-aws-s3-bucket-do-not-work)
Since there is no actual S3 object, BrowserRouter causes NoSuchKeyErrors when reloading a page. 
<img width="806" alt="Screenshot 2024-01-18 at 8 52 52 am" src="https://github.com/aws-samples/aws-genai-llm-chatbot/assets/11172727/06c0e0d3-e1c3-4dea-9fa9-5ff7c656e023">

Potential fixes to not have "ugly" Hash pathing (i.e chatbot-dev.kmanuwai.test/index.html#/chatbot/playground):
1. Run an if else on whether to use a HashRouter or BrowserRouter version of the app here: [https://github.com/aws-samples/aws-genai-llm-chatbot/blob/a8c6709c72e9600916b1c36f[…]/lib/user-interface/react-app/src/components/app-configured.tsx](https://github.com/aws-samples/aws-genai-llm-chatbot/blob/a8c6709c72e9600916b1c36fab4996827a2bcc2f/lib/user-interface/react-app/src/components/app-configured.tsx#L159) although this does not feel great having this happen every time the app is loaded?

2. Likely the preferred option - have two versions of [app.tsx](https://github.com/aws-samples/aws-genai-llm-chatbot/blob/a8c6709c72e9600916b1c36fab4996827a2bcc2f/lib/user-interface/react-app/src/app.tsx#L23), one with HashRouter, another with BrowserRouter. And deploy only the relevant one during the S3 asset[ deployment](https://github.com/aws-samples/aws-genai-llm-chatbot/blob/a8c6709c72e9600916b1c36fab4996827a2bcc2f/lib/user-interface/index.ts#L239). However, I am unsure how best to achieve this

3. Use Option 2 suggested here: https://stackoverflow.com/questions/66326106/react-router-locations-on-aws-s3-bucket-do-not-work. This uses Hashrouter but cleans up the hash in the browser url.

Reply:
What about setting in S3 `Error Document` to point to `index.html`? 

You still get the same behaviour?

Reply:
I did research this as an option but it did not work in this scenario unfortunately. How best do you think it is to proceed then?

Reply:
@bigadsoleiman I have updated this with a conditional on which Router to use based on the config.json privateWebsite flag. Let me know if you have any issues with this.
https://github.com/aws-samples/aws-genai-llm-chatbot/pull/310/commits/b770d2d4beae3b190e009e5730ccb5dab0c6702d?diff=split&w=0#diff-e5e3c6e8aced16a21c21f373c1b969eda18a25a2a3e4319fa3ba576fb2a77368R22